lib/membrane_wav/serializer.ex

```diff
@@ -77,20 +95,33 @@ defmodule Membrane.WAV.Serializer do
         buffers_count,
         :buffers,
         context,
-        %{header_created: true, frames_per_buffer: frames} = state
+        %{frames_per_buffer: frames} = state
       ) do
     caps = context.pads.output.caps
     demand_size = Caps.frames_to_bytes(frames, caps) * buffers_count
     {{:ok, demand: {:input, demand_size}}, state}
   end
 
   @impl true
-  def handle_process(:input, buffer, _context, %{header_created: true} = state) do
-    {{:ok, buffer: {:output, buffer}, redemand: :output}, state}
+  def handle_process_list(:input, _buffers, _context, %{header_length: 0}) do
+    raise(RuntimeError, "buffers received before caps, so the header is not created yet")
   end
 
-  def handle_process(:input, _buffer, _context, %{header_created: false}) do
-    raise(RuntimeError, "buffer received before caps, so the header is not created yet")
+  def handle_process_list(:input, buffers, _context, %{data_length: data_length} = state) do
+    state =
+      buffers
+      |> Enum.reduce(data_length, fn %Buffer{payload: payload}, acc ->
+        acc + byte_size(payload)
+      end)
+      |> then(&Map.put(state, :data_length, &1))
```

Comment:
A bit of nitpicking, but I don't like when, while piping, the "subject" changes. I'd split that:
```suggestion
    data_length =
      buffers
      |> Enum.reduce(data_length, fn %Buffer{payload: payload}, acc ->
        acc + byte_size(payload)
      end)
    
    state = %{state | data_length: data_length}
```